### PR TITLE
Triangles and Transitivity

### PR DESCRIPTION
**Pull request by egolol**

### CODE REVIEW

File: braph2genesis/measures/_Transitivity.gen.m [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+%%%% ¡calculate!
+g = m.get('G');  % graph from measure class
+A = g.get('A');  % cell array with adjacency matrix 
+
+L = g.layernumber();
+
+triangles = calculateValue@Triangles(m, prop);
+
+transitivity = cell(g.layernumber(), 1);
+directionality_type =  g.getDirectionalityType(g.layernumber());
```
Comment (annacanal) [inline braph2genesis/measures/_Transitivity.gen.m:40]: use L variable, that you declared on line 35

File: braph2genesis/measures/_Triangles.gen.m [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+%%% ¡compatible_graphs!
+GraphWU
+GraphWD
+GraphBU
+GraphBD
+MultiplexGraphWU
+MultiplexGraphWD
+MultiplexGraphBU
+MultiplexGraphBD
+
```
Comment (annacanal) [inline braph2genesis/measures/_Triangles.gen.m:30]: one extra space line -> remove

File: braph2genesis/measures/_Triangles.gen.m [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+%% ¡constants!
+RULES = { ... % vector of triangles rules
+    'all' ...
+    'middleman' ...
+    'in' ...
+    'out' ...
+    'cycle' ...
+    };
```
Comment (annacanal) [inline braph2genesis/measures/_Triangles.gen.m:39]: shouldnt we explain using comments the meaning of these rules? as we had in braph2?

File: braph2genesis/measures/_Triangles.gen.m [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+5
+
+%% ¡props_update!
+
+%%% ¡prop!
+M (result, cell) is the triangles.
+%%%% ¡calculate!
+g = m.get('G');  % graph from measure class
+A = g.get('A');  % cell array with adjacency matrix 
+L = g.layernumber();
```
Comment (annacanal) [inline braph2genesis/measures/_Triangles.gen.m:56]: not using L later.

**Review by annacanal — CHANGES_REQUESTED** [review]
Also merge the current gv-genesis-2 with the new format of multiplex graphs. you will have to adapt the tests

**Review by c-yuwei — APPROVED** [review]

File: braph2genesis/measures/_Triangles.gen.m [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+g = GraphBU('B', B);
+triangles = Triangles('G', g);
+
+assert(isequal(triangles.get('M'), known_triangles), ...
+    [BRAPH2.STR ':Triangles:' BRAPH2.BUG_ERR], ...
+    'Triangles is not being calculated correctly for GraphBU.')
+
+
+%% Test 2: GraphBD
```
Comment (annacanal) [inline braph2genesis/measures/_Triangles.gen.m:112]: adapt the format of test 2

File: braph2genesis/measures/_Transitivity.gen.m [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    };
+
+known_transitivity = {
+                 3/4
+                 3/4
+                 };      
+
+g = MultiplexGraphBU('B', A);
+transitivity = Transitivity('G', g);
+
```
Comment (annacanal) [inline braph2genesis/measures/_Transitivity.gen.m:114]: Adapt to the new format of Multiplex graphs,
It fails the test now for MultiplexGraphBU

**Review by annacanal — CHANGES_REQUESTED** [review]

**Review by annacanal — APPROVED** [review]

File: braph2genesis/measures/_Triangles.gen.m [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+    'cycle' ...
+    };
+
+%% ¡props!
+%%% ¡prop! 
+rule (metadata, SCALAR) 
+%%%% ¡check_prop!
+check =  value >= 1 && value <= 5;
+%%%% ¡default!
+5
```
Comment (giovannivolpe) [inline braph2genesis/measures/_Triangles.gen.m:46]: use format = option instead

**Review by giovannivolpe — CHANGES_REQUESTED** [review]
Use format option for the rules

**Review by giovannivolpe — APPROVED** [review]

**Pull request merged**